src/filters/frame_skipping_pp.py

```diff
@@ -0,0 +1,101 @@
+from src.filters.abstract_pp import AbstractPP
+from src.utils import image_utils
+from typing import List
+from enum import Enum, unique
+
+
+@unique
+class DistanceMetric(Enum):
+    ABSOLUTE_DIFFERENCE = "absolute_difference"
+    MSE_DIFFERENCE = "mse_difference"
+
+
+class FrameSkippingPP(AbstractPP):
+    """
+    Class that performs the task of frame skipping. 
+    Frame skipping can be done in two ways:
+        1. Compare all pixels of a frame
+        2. Compare only foreground pixels.
+    """
+
+    def __init__(self, threshold=0.0, compare_foreground=False, 
+                 distance_metric=DistanceMetric.ABSOLUTE_DIFFERENCE.value):
+        """
+        Constructor for this class.
+
+        :param threshold: threshold value for difference
+                          below which frames are skipped
+        :param compare_foreground: boolean value to indicate if only 
+                                   foreground pixels are to be compared.
+        :param distance_metric: DistanceMetric ENUM attribute
+        :return None
+        """        
+        self.threshold = threshold
+        self.compare_foreground = compare_foreground
+        self.distance_metric = distance_metric
+
+    def predict(self, batch) -> List[bool]:
+        """
+        Function that predicts whether a frame needs to be skipped or not
+        based on the threshold value set.
+
+        :param batch: an EVA batch of EVA frames
+        :return List of booleans that indicates whether the frame at a given
+                index should be skipped or not.
+                False => do not skip frame
+                True => skip frame
+        """
+        prev_frame = None
+        skip_frames = []
+        frames = batch.frames_as_numpy_array()
+        for frame in frames:
+            if prev_frame is not None:
+                """
+                If compare_foreground set to true, calculate distance
+                metric on only the foreground pixels
+                else on the entire image.
+                """
+                if self.compare_foreground is True:
```

Comment:
This logic is not tested.

Reply:
testing this logic using mse and absolute difference